matrix-sdk-android/src/test/java/org/matrix/android/sdk/internal/database/mapper/LiveLocationShareAggregatedSummaryMapperTest.kt

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2022 The Matrix.org Foundation C.I.C.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.matrix.android.sdk.internal.database.mapper
+
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.mockkObject
+import io.mockk.unmockkObject
+import org.amshove.kluent.internal.assertEquals
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.matrix.android.sdk.api.session.events.model.Content
+import org.matrix.android.sdk.api.session.room.model.livelocation.LiveLocationShareAggregatedSummary
+import org.matrix.android.sdk.internal.database.model.livelocation.LiveLocationShareAggregatedSummaryEntity
+
+class LiveLocationShareAggregatedSummaryMapperTest {
+
+    private val mapper = LiveLocationShareAggregatedSummaryMapper()
+
+    @Before
+    fun setUp() {
+        mockkObject(ContentMapper)
+    }
+
+    @After
+    fun tearDown() {
+        unmockkObject(ContentMapper)
+    }
+
+    @Test
+    fun `given an entity then result should be mapped correctly`() {
+        val userId = "userId"
+        val timeout = 123L
+        val isActive = true
+        val lastKnownLocationContent = "lastKnownLocationContent"
+        val entity = LiveLocationShareAggregatedSummaryEntity(
+                userId = userId,
+                isActive = isActive,
+                endOfLiveTimestampMillis = timeout,
+                lastLocationContent = lastKnownLocationContent
+        )
+        val content = mockk<Content>()
+        every { ContentMapper.map(lastKnownLocationContent) } returns content
+
+        val summary = mapper.map(entity)
+
+        // note: unfortunately the implementation relies on an inline method to map the lastLocationDataContent
```

Comment:
for this case I would lean towards calling into the real `ContentMapper`, especially as that's where the majority of the logic of the summary mapper sits, it also means the test can be simplified! 

```kotlin
private const val ANY_USER_ID = "a-user-id"
private const val ANY_ACTIVE_STATE = true
private const val ANY_TIMEOUT = 123L
private val A_LOCATION_INFO = LocationInfo("a-geo-uri")

class LiveLocationShareAggregatedSummaryMapperTest {

    private val mapper = LiveLocationShareAggregatedSummaryMapper()

    @Test
    fun `given an entity then result should be mapped correctly`() {
        val entity = anEntity(content = MessageBeaconLocationDataContent(locationInfo = A_LOCATION_INFO))

        val summary = mapper.map(entity)

        summary shouldBeEqualTo LiveLocationShareAggregatedSummary(
                userId = ANY_USER_ID,
                isActive = ANY_ACTIVE_STATE,
                endOfLiveTimestampMillis = ANY_TIMEOUT,
                lastLocationDataContent = MessageBeaconLocationDataContent(locationInfo = A_LOCATION_INFO)
        )
    }

    private fun anEntity(content: MessageBeaconLocationDataContent) = LiveLocationShareAggregatedSummaryEntity(
            userId = ANY_USER_ID,
            isActive = ANY_ACTIVE_STATE,
            endOfLiveTimestampMillis = ANY_TIMEOUT,
            lastLocationContent = Moshi.Builder().build().adapter(MessageBeaconLocationDataContent::class.java).toJson(content)
    )
}
```

Reply:
Thanks for the suggested code! I will update it.

Reply:
See [f707f17](https://github.com/vector-im/element-android/pull/6129/commits/f707f177d923623b7afa0f4201756614e29224e9)